package/MDAnalysis/coordinates/base.py

```diff
@@ -592,6 +592,49 @@ def volume(self):
         """volume of the unitcell"""
         return core.box_volume(self.dimensions)
 
+    @property
+    def dt(self):
+        """The time difference in ps between timesteps
+
+        .. Note::
+           This defaults to 1.0 ps in the absence of time data
+
+        .. versionadded:: 0.11.0
+        """
+        try:
+            return self.data['dt']
+        except KeyError:
+            warnings.warn("Reader has no dt information, set to 1.0 ps")
+            return 1.0
+
+    @dt.setter
+    def dt(self, new):
+        self.data['dt'] = new
+
+    @property
+    def time(self):
+        """The time in ps of this timestep
+
+        This is calculated as::
+
+          time = data['time_offset'] + data['time']
+
+        Or, if the trajectory doesn't provide time information::
+
+          time = data['offset'] + frame * dt
+
+        .. versionadded:: 0.11.0
+        """
+        offset = self.data.get('time_offset', 0)
+        try:
+            return self.data['time'] + offset
+        except KeyError:
+            return self.dt * self.frame + offset
+
+    @time.setter
+    def time(self, new):
+        self.data['time'] = new
```

Comment:
So `Timestep.data` only contains values that are either set by the user or read from the trajectory itself? 

Any top level attributes such as `Timestep.dt` are either taken from `data` or computed on the fly, without caching so that user changes take effect immediately?